src/eval.ml

```diff
@@ -199,22 +212,23 @@ and tree_walk : dtree -> stack -> (term * stack) option = fun tree stk ->
           | CondNL(i, j) ->
               if eq_modulo vars.(i) vars.(j) then ok else fail
           | CondFV(xs,i) ->
-              let xs = Array.map (fun e -> VarMap.find e fresh_vars) xs in
+              let xs = Array.map (fun e -> IntMap.find e id_vars) xs in
+              let fn b =
```

Comment:
Try to use more informative names like:
```
              let id_vars_notin b =
                let fn _ x = not (Bindlib.occur x b) in IntMap.for_all fn id_vars
```

Reply:
But the `if ... then ... else` won't be a one liner

Reply:
Perhaps but understandability is more important.